remote.go

```
@@ -90,6 +101,50 @@ func NewRegistryClient(insecure bool) *RegistryClient {
 	return &RegistryClient{http.Client{Transport: tr}}
 }
 
+func (r *RegistryClient) SignRequest(info *RepoInfo, request *http.Request) {
+	if info.Host == "iad.ocir.io" {
+		// Set the Date header
+		request.Header.Set("Date", time.Now().UTC().Format(http.TimeFormat))
+		// And a provider of cryptographic keys
+		provider := common.DefaultConfigProvider()
+		// Mandatory headers to be used in the sign process
+		defaultGenericHeaders := []string{"date", "(request-target)", "host"}
+
+		// Optional headers
+		optionalHeaders := []string{"content-length", "content-type", "x-content-sha256"}
+
+		// A predicate that specifies when to use the optional signing headers
+		optionalHeadersPredicate := func(re *http.Request) bool {
+			return re.Method == http.MethodPost
+		}
+		// Build the signer
+		signer := common.RequestSignerWithBodyHashingPredicate(provider, defaultGenericHeaders, optionalHeaders, optionalHeadersPredicate)
+		// Sign the request
+		tenancyId, _ := provider.TenancyOCID()
+		logrus.Debugf("Signing request %s", tenancyId)
+		signer.Sign(request)
+	}
+}
+
+func getResponseMessage(response *http.Response) string {
+	var buf bytes.Buffer
+	if _, err := buf.ReadFrom(response.Body); err != nil {
+		logrus.Warnf("Failed to read response body: %v", err)
```

Definitely shouldn't be swallowing an error here - make it bubble or make the logrus fatal.